terminus-ui/src/select/select.component.spec.ts

```diff
@@ -1,192 +1,1537 @@
+        fixture.detectChanges();
+
+        // move down to Florida and try to select it
+        dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+        dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+        dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+        dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+        dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+        dispatchKeyboardEvent(input, 'keydown', ENTER);
+        fixture.detectChanges();
+
+        // Verify the selection did NOT work
+        chips = getAllChipInstances(fixture)!;
+        expect(chips.length).toEqual(1);
+        expect(fixture.componentInstance.duplicate).toHaveBeenCalledWith('Florida');
+
+        expect.assertions(3);
+      });
+
+
+      describe(`when allowed`, () => {
+
+        test(`should allow a duplicate selection`, () => {
+          const fixture = createComponent(testComponents.SeededAutocomplete);
+          fixture.componentInstance.allowDuplicates = true;
+          fixture.detectChanges();
+
+          let chips = getAllChipInstances(fixture)!;
+          expect(chips.length).toEqual(1);
+
+          const input = getAutocompleteInput(fixture)!;
+          typeInElement('fl', input!);
+          fixture.detectChanges();
+
+          // move down to Florida and try to select it
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', ENTER);
+          fixture.detectChanges();
+
+          // Verify the selection DID work
+          chips = getAllChipInstances(fixture)!;
+          expect(chips.length).toEqual(2);
+          expect(getSelectInstance(fixture).autocompleteSelections).toEqual(['Florida', 'Florida']);
+
+          expect.assertions(3);
+        });
+
+      });
+
+    });
+
+
+    describe(`trigger`, () => {
+
+      describe(`in single selection mode`, () => {
+
+        test(`should set single value`, () => {
+          const fixture = createComponent(testComponents.SeededAutocomplete);
+          fixture.componentInstance.allowMultiple = false;
+          fixture.componentInstance.keepOpen = false;
+          fixture.detectChanges();
+          const instance = getSelectInstance(fixture);
+          const input = getAutocompleteInput(fixture)!;
+
+          typeInElement('fl', input);
+          fixture.detectChanges();
+
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', DOWN_ARROW);
+          dispatchKeyboardEvent(input, 'keydown', ENTER);
+          fixture.detectChanges();
+
+          expect(instance.autocompleteFormControl.value).toEqual(['Arkansas']);
```

Comment:
Because our test components don't actually have any search/filtering built it. Since it is not code from the library, there is no reason to test it.

So in normal use:

- user types query
- consumer fetches a new collection matching the query
- component shows collection
- user navigates and selects

In our tests:

- we fake a user query
- we show the exact same collection no matter the query
- we fake user navigation to an item and select

So there is no actual relation between our fake query and the results shown. We just have to keep remembering that unlike most autocomplete components, we are not doing any local filtering.